### PR TITLE
Paas 1164 fix onAfterClone's haproxy behaviour

### PR DESCRIPTION
**Pull request by maxenceA**
JIRA issue: https://jira.jahia.org/browse/PAAS-1164

Short description: onAfterClone event now refresh the proc node haproxy backend.

### CODE REVIEW

File: onAfterClone.yml [hunk 1]
```diff
@@ -54,11 +54,20 @@ actions:
...
         cmd[${this}]: |-
           echo "    server ${env.appid}-${@i.id} ${@i.intIP}:80 check cookie s${@i.id}" >> /etc/haproxy/haproxy.cfg.d/10-jahia.cfg
           sudo service haproxy reload
         user: haproxy
+    - forEach(nodes.proc):
+        cmd[${this}]: |-
+          echo "    server ${env.appid}-${@i.id} ${@i.intIP}:80 check cookie s${@i.id}" >> /etc/haproxy/haproxy.cfg.d/11-proc.cfg
+          sed -ie '/default_backend bk_jahia/i\    use_backend proc if { cook(JSESSIONID),lower -m beg s${@i.id} }' /etc/haproxy/haproxy.cfg.d/00-global.cfg
+          sudo service haproxy reload
+        user: haproxy
```
Comment (Ronan-Gorain) [inline onAfterClone.yml:69]: I am really sorry but I didn't really think about the changes you showed me earlier... 

There is no need to iterate on processing nodes cause there is always one. 
Two little sed to search and replace the proccessing node id should do the trick here. 
And if you place the seds after the first forEach you can remove the haproxy reload an place it just after your sed (cause it was stupid to reload it multiple times)

Once again, sorry for not having checked correctly.

Comment (maxenceA) [reply onAfterClone.yml:69]: No worries, I'll update the branch with the changes!

**Review by laurentfufu — COMMENTED** [review]
Maybe wrong but for what I understand, don't like this part sorry :(
```yaml
actions:
  resetHaproxyBackends:
    - cmd[bl]: |-
        sed -ie "/$server.*:80 check cookie/d" /etc/haproxy/haproxy.cfg.d/10-jahia.cfg
    - forEach(nodes.cp):
        cmd[${this}]: |-
          echo "    server ${env.appid}-${@i.id} ${@i.intIP}:80 check cookie s${@i.id}" >> /etc/haproxy/haproxy.cfg.d/10-jahia.cfg
    - cmd[bl]: |-
        sed -ie 's/^.*server.*:80 check cookie.*$/\    server ${env.appid}-${nodes.proc.first.id} ${nodes.proc.first.intIP}:80 check cookie s${nodes.proc.first.id}/' /etc/haproxy/haproxy.cfg.d/11-proc.cfg
        sed -ie 's/^.*use_backend proc if { cook(JSESSIONID),lower -m beg.*$/\    use_backend proc if { cook(JSESSIONID),lower -m beg s${nodes.proc.first.id} }/' /etc/haproxy/haproxy.cfg.d/00-global.cfg
        sudo service haproxy reload
      user: haproxy
```
sed lines are too long and not visual at all to me.

IMHO, something like this will be smoother to read and understand:
```yaml
actions:
  resetHaproxyBackends:
    - cmd[bl]: |-
        sed -e "/$server.*:80 check cookie/d" \
            -e '/^.*use_backend proc if { cook(JSESSIONID),lower -m beg.*$/d' \
            -i /etc/haproxy/haproxy.cfg.d/{10-jahia,11-proc}.cfg
        echo "    server ${env.appid}-${nodes.proc.first.id} ${nodes.proc.first.intIP}:80 check cookie s${nodes.proc.first.id}" \
            >> /etc/haproxy/haproxy.cfg.d/11-proc.cfg
        echo "    use_backend proc if { cook(JSESSIONID),lower -m beg s${nodes.proc.first.id} }" \
            >> /etc/haproxy/haproxy.cfg.d/00-global.cfg
    - forEach(nodes.cp):
        cmd[${this}]: |-
          echo "    server ${env.appid}-${@i.id} ${@i.intIP}:80 check cookie s${@i.id}" >> /etc/haproxy/haproxy.cfg.d/10-jahia.cfg
          sudo service haproxy reload
        user: haproxy
```

**Comment by maxenceA** [comment]
I tried to make these lines more readable.
I kept a sed for 00-global.cfg because i didn't want the use_backend condition to be at the end of the file.

```YAML
actions:
  resetHaproxyBackends:
    - cmd[bl]: |-
        sed -e "/$server.*:80 check cookie/d" \
            -i /etc/haproxy/haproxy.cfg.d/{10-jahia,11-proc}.cfg
        echo "    server ${env.appid}-${nodes.proc.first.id} ${nodes.proc.first.intIP}:80 check cookie s${nodes.proc.first.id}" \
            >> /etc/haproxy/haproxy.cfg.d/11-proc.cfg
        sed -e 's/cook(JSESSIONID),lower -m beg s.*/cook(JSESSIONID),lower -m beg s${nodes.proc.first.id} }/' \
            -i /etc/haproxy/haproxy.cfg.d/00-global.cfg
    - forEach(nodes.cp):
        cmd[${this}]: |-
          echo "    server ${env.appid}-${@i.id} ${@i.intIP}:80 check cookie s${@i.id}" >> /etc/haproxy/haproxy.cfg.d/10-jahia.cfg
          sudo service haproxy reload
        user: haproxy`
```

**Review by laurentfufu — APPROVED** [review]

**Pull request merged**